### ISSUE TITLE
Low accuracy with the flower dataset?

### ISSUE THREAD

**Issue by GUIMINLONG** [jiaowoguanren0615/FasterNet-Pytorch#1]
Hi, thanks for sharing this project. I am try to train a Fasternet model with my own dataset. This  easy using project is what I am exact need.
I have some problem when try to run this project:
(1) Low accuracy for flower dataset.  I have tried with all the different size model, only accuracy around 45~50% were achieved. Is this normal?  I am confused that a claimed outstanding model with only accuracy like this.

(2) Besides, when I try to train the "Fasternet_M" model, the model will reach to max accuracy at the first or second epoch, then the accuracy decrease until the program is stopped because  "Loss value is Nan". Could please explain the potential reason of this problem? 

regards,
LONG

**Comment by jiaowoguanren0615** [comment]
Well, the first question, if you train this model without pre-trained weights, it's normal, you can download the pre-trained weights in [https://github.com/JierunChen/FasterNet/?tab=readme-ov-file#3-pre-trained-models], after downloading, you must modify the optimizer in train_gpu.py, just use Adam, lr=0.001. In addition, you should delete the head weights.

The second question, I've never had this problem training my own dataset, do you use the flower dataset or your own dataset.
Here is example code for transferring learning:

```
    if args.finetune:
        if args.finetune.startswith('https'):
            checkpoint = torch.hub.load_state_dict_from_url(
                args.finetune, map_location='cpu', check_hash=True)
        else:
            checkpoint = utils.load_model(args.finetune, model)

        checkpoint_model = checkpoint
        # state_dict = model.state_dict()
        for k in list(checkpoint_model.keys()):
            if 'head' in k:
                print(f"Removing key {k} from pretrained checkpoint")
                del checkpoint_model[k]
```

**Comment by GUIMINLONG** [comment]
> The second question, I've never had this problem training my own dataset, do you use the flower dataset or your own dataset.
> Here is example code for transferring learning:

Thanks for your prompt response, and detail explanation.   

Currently I am trying to run through this project, therefore, all training work is done with flower recognition dataset.
Could you please sharing your training result of different size model? For example, with Fasternet_t0 and Flower recognition dataset, what is the highest accuracy we can achieve?  I just want to confirm if I have set everything in right way, and get close to its best performance.

Best regards

**Comment by jiaowoguanren0615** [comment]
Sorry, because the flower dataset is not a standard competition dataset, not like the ImageNet or other competition datasets. So, I just ensured that the code would not run bugs again and did not train for many rounds until a very good accuracy or ROC was achieved. Moreover, the official team obtain the best performance in ImageNet-1K by using **Fasternet_L** model is **83.5%**, **FasterNet_t0** achieves the **71.9%** of accuracy in ImageNet-1K.

**Comment by jiaowoguanren0615** [comment]
If you want to obtain better performance in classification tasks, you can use my another model named  UniRepLKNet [https://github.com/jiaowoguanren0615/UniRepLKNet-Pytorch], I achieve **92% of accuracy and 99% of ROC** in the flower dataset with training 5 epochs. Just download the pre-trained weights and modify the optimizer in train_gpu.py

```
    if args.finetune:
        if args.finetune.startswith('https'):
            checkpoint = torch.hub.load_state_dict_from_url(
                args.finetune, map_location='cpu', check_hash=True)
        else:
            checkpoint = utils.load_model(args.finetune, model)

        # checkpoint_model = checkpoint['model']
        checkpoint_model = checkpoint
        # state_dict = model.state_dict()
        for k in list(checkpoint_model.keys()):
            if 'head' in k:
                print(f"Removing key {k} from pretrained checkpoint")
                del checkpoint_model[k]
```

**Comment by GUIMINLONG** [comment]
> If you want to obtain better performance in classification tasks, you can use my another model named UniRepLKNet [https://github.com/jiaowoguanren0615/UniRepLKNet-Pytorch], I achieve **92% of accuracy and 99% of ROC** in the flower dataset with training 5 epochs. Just download the pre-trained weights and modify the optimizer in train_gpu.py
> 
> ```
>     if args.finetune:
>         if args.finetune.startswith('https'):
>             checkpoint = torch.hub.load_state_dict_from_url(
>                 args.finetune, map_location='cpu', check_hash=True)
>         else:
>             checkpoint = utils.load_model(args.finetune, model)
> 
>         # checkpoint_model = checkpoint['model']
>         checkpoint_model = checkpoint
>         # state_dict = model.state_dict()
>         for k in list(checkpoint_model.keys()):
>             if 'head' in k:
>                 print(f"Removing key {k} from pretrained checkpoint")
>                 del checkpoint_model[k]
> ```

OK， very appreciate for your reply. Thanks
